### PR TITLE
Rector for adding latte {varType}s based on presenter template parameters

### PR DESCRIPTION
**Pull request by lulco**

### CODE REVIEW

File: src/Rector/Class_/LatteVarTypesBasedOnPresenterTemplateParametersRector.php [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+        $this->printVarTypesToTemplateFiles($actionVarTypes, $presenterName);
+
+        return null;
+    }
+
+    /**
+     * @return LatteVariableType[]
+     */
+    private function findVarTypesForAction(ClassMethod $method): array
```
Comment (TomasVotruba) [inline src/Rector/Class_/LatteVarTypesBasedOnPresenterTemplateParametersRector.php:125]: Could you extract this to own service? It looks like complex logic

File: phpstan.neon [hunk 2]
```diff
@@ -30,6 +30,8 @@ parameters:
 
     ignoreErrors:
         - '#Class "Rector\\Nette\\Rector\\Class_\\MoveFinalGetUserToCheckRequirementsClassMethodRector" is missing @see annotation with test case class reference#'
+        # We need to think how to test rectors like this one
+        - '#Class "Rector\\Nette\\Rector\\Class_\\LatteVarTypesBasedOnPresenterTemplateParametersRector" is missing @see annotation with test case class reference#'
```
Comment (TomasVotruba) [inline phpstan.neon:34]: Test is needed

Comment (lulco) [reply phpstan.neon:34]: Sure thing :) my question is "how to test rectors which change different file(s)? Maybe they create some file and change the original file too.". Is there any engine for that in rector(-src)?

File: src/Rector/Class_/LatteVarTypesBasedOnPresenterTemplateParametersRector.php (hunk 1 above)
Comment (lulco) [reply src/Rector/Class_/LatteVarTypesBasedOnPresenterTemplateParametersRector.php:125]: It's planned in some future PR. I want to create following implementation for Controls and I will see what is common.

File: phpstan.neon (hunk 2 above)
Comment (TomasVotruba) [reply phpstan.neon:34]: I see. There is a test case that checks the dumped content, but not changed external file.

I would simply compare external content before and after + make a backup of original file to restore it in the end. Filesystem + read/copy/write combo

Comment (lulco) [reply phpstan.neon]: I've added some test, check it please :)

**Comment by TomasVotruba** [comment]
Thank you! This is very helpful addition, I don't anyone expect Rector to be that smart :) 

Great job!

**Comment by lulco** [comment]
and if everything will work as I expect, Rector will be even smarter ;)

**Pull request merged**